README.md

```diff
@@ -2,139 +2,140 @@
 
 # Nyaml to nxdl
 
-A tool to convert yaml NeXus application definitions (nyaml) to nexus definitions language (nxdl) and vice versa.
+Nyaml to Nxdl is a versatile tool designed for converting NeXus application definitions from YAML (nyaml) to the Nexus Definitions Language (nxdl) format and vice versa. This command-line interface offers a simple and efficient way to work with NeXus data definitions.
 
-This tool is a simple command line interface and can be used by calling `nyaml2nxdl` or its shortcut `n2n`:
+## Installation
 
-```console
+You can easily install the tool using `pip`:
+
+```bash
+pip install nyaml
+```
+
+Alternatively, you can install it as a development package from the GitHub repository:
+
+```bash
+git clone https://github.com/FAIRmat-NFDI/nyaml.git
+cd nyaml
+pip install -e ".[dev]"
+```
+
+We also provide a pre-commit hook for code formatting and linting, which you can install with:
+
+```bash
+pre-commit install
+```
+
+## Usage
+
+The primary commands for using nyaml are `nyaml2nxdl` and its shortcut, `n2n`. Here is an overview of the available options:
+
+```bash
 user@box:~$ nyaml2nxdl --help
 Usage: nyaml2nxdl [OPTIONS] INPUT_FILE
 
   Main function that distinguishes the input file format and launches the
   tools.
 
 Options:
-  --output-file TEXT   The output file path to write the converted file to
-  --check-consistency  Check if yaml and nxdl can be converted from one to
-                       another version recursively and get the same version of
-                       file. E.g. from NXexample.nxdl.xml to
-                       NXexample_consistency.nxdl.xml.
-  --do-not-store-nxdl  Whether the input nxdl file will be stored as a comment
-                       at the end of output yaml file.
-  --verbose            Print in standard output keywords and value types to
-                       help possible issues in yaml files
+  --output-file TEXT   Specify the output file path for the converted file.
+  --check-consistency  Check whether YAML and NXDL can be recursively
+                       converted, ensuring version consistency.
+  --do-not-store-nxdl  Prevent the input NXDL file from being stored as a
+                       comment at the end of the output YAML file.
+  --verbose            Display keywords and value types in standard output to
+                       assist in identifying issues in YAML files.
```

Comment:
add `--append` here

Reply:
There is no append keyword in the cli command line arguments. This part here is directly generated from the cli help string and not manually created.

Reply:
Right, but then how does this work:

Additionally, if the --append argument is provided, the input XML or YAML file is interpreted as an extension of a base class. Entries from the input are appended below a standard NeXus base class. You need to specify both your input file (with YAML or XML extension) and the NeXus class (with no extension). Both the .yaml and .nxdl.xml files of the extended class are generated.

Reply:
🤔 Did we remove this at some point?

Reply:
I removed it the documentation part. It just seemed as an old residual.